### PR TITLE
Pulsar 2.8.4 release notes

### PR DESCRIPTION
**Pull request by BewareMyPower**
image

image

image

image

image

image

image

image

image

image

### CODE REVIEW

File: site2/website-next/release-notes/versioned/pulsar-2.8.4.md [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+- [fix][broker] Fix passing incorrect authentication data [16840](https://github.com/apache/pulsar/pull/16840)
+- [fix][broker] ManagedCursor: mark delete no callback when create meta-ledger fail [16841](https://github.com/apache/pulsar/pull/16841)
+- [improve][broker] Avoid reconnection when a partitioned topic was created concurrently [16867](https://github.com/apache/pulsar/pull/16867)
+- [improve][broker] Recycle OpReadEntry in some corner cases [16869](https://github.com/apache/pulsar/pull/16869)
+- [fix][broker] Fix compaction subscription acknowledge Marker msg issue. [16918](https://github.com/apache/pulsar/pull/16918)
+
+#### Security
+- [fix][sec] Use provider's canLookupAsync method for AuthorizationService [11777](https://github.com/apache/pulsar/pull/11777)
+- [improve][sec] Optimize the logic of allowing namespace operation [13090](https://github.com/apache/pulsar/pull/13090)
+- [improve][sec] Full-support set ssl provider, ciphers and protocols [13740](https://github.com/apache/pulsar/pull/13740)
```
Comment (Anonymitaet) [inline site2/website-next/release-notes/versioned/pulsar-2.8.4.md:74]:
```suggestion
- [improve][sec] Full-support set SSL provider, ciphers and protocols [13740](https://github.com/apache/pulsar/pull/13740)
```

File: site2/website-next/release-notes/versioned/pulsar-2.8.4.md [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+
+#### Proxy
+- [improve][proxy] Log warning when opening connection to broker fails [14710](https://github.com/apache/pulsar/pull/14710)
+- [refactor][proxy] Refactor Proxy code and fix connection stalling by switching to auto read mode [14713](https://github.com/apache/pulsar/pull/14713)
+- [improve][proxy] Configure Netty DNS resolver to match JDK DNS caching setting, share DNS resolver instance in Proxy [15219](https://github.com/apache/pulsar/pull/15219)
+- [fix][proxy] Fix proxy connection leak when inbound connection closes while connecting is in progress [15366](https://github.com/apache/pulsar/pull/15366)
+- [fix][proxy] Fix DNS server denial-of-service issue when DNS entry expires [15403](https://github.com/apache/pulsar/pull/15403)
+- [improve][proxy] Remove unnecessary blocking DNS lookup in LookupProxyHandler [15415](https://github.com/apache/pulsar/pull/15415)
+- [fix][proxy] Do not preserve host when forwarding admin requests. [16342](https://github.com/apache/pulsar/pull/16342)
+- [fix][proxy] Fix client service url [16834](https://github.com/apache/pulsar/pull/16834)
```
Comment (Anonymitaet) [inline site2/website-next/release-notes/versioned/pulsar-2.8.4.md:137]:
```suggestion
- [fix][proxy] Fix client service URL [16834](https://github.com/apache/pulsar/pull/16834)
```

**Review by Anonymitaet — APPROVED** [review]

File: site2/website-next/release-notes/versioned/pulsar-2.8.4.md [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+- [cleanup][broker] Fix typo in enum name and handle closing of the channel properly since writeAndFlush is asynchronous [15384](https://github.com/apache/pulsar/pull/15384)
+- [fix][broker] Avoid heartbeat topic to offload. [15393](https://github.com/apache/pulsar/pull/15393)
+- [fix][broker] Fix deadlock in broker after race condition in topic creation failure [15570](https://github.com/apache/pulsar/pull/15570)
+- [fix][broker] fix calculation in getNumberOfEntriesInStorage [15627](https://github.com/apache/pulsar/pull/15627)
+- [fix][broker] Fix NPE when set `AutoTopicCreationOverride` [15653](https://github.com/apache/pulsar/pull/15653)
+- [fix][broker] Fix creating producer failure when set backlog quota [15663](https://github.com/apache/pulsar/pull/15663)
+- [fix][broker] Fix NPE when put value to `RangeCache` [15707](https://github.com/apache/pulsar/pull/15707)
+- [fix][broker] Fix can not enable system topic if `AutoUpdateSchemaEnabled=false` [15759](https://github.com/apache/pulsar/pull/15759)
+- [fix][broker] Fix wrong unit of NIC speed on linux  [15770](https://github.com/apache/pulsar/pull/15770)
```
Comment (momo-jun) [inline site2/website-next/release-notes/versioned/pulsar-2.8.4.md:54]:
```suggestion
- [fix][broker] Fix wrong unit of NIC speed on Linux  [15770](https://github.com/apache/pulsar/pull/15770)
```

File: site2/website-next/release-notes/versioned/pulsar-2.8.4.md [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+- [fix][fn] fix some IOExceptions when create functions from package URL [14553](https://github.com/apache/pulsar/pull/14553)
+- [fix][fn] handle NPE when `getLeader` returns null [15058](https://github.com/apache/pulsar/pull/15058)
+- [improve][fn] Add KeyStore support in WebSocket, Function Worker HTTPS Servers  [15084](https://github.com/apache/pulsar/pull/15084)
+- [fix][function] Check executor null when close the FileSource [15247](https://github.com/apache/pulsar/pull/15247)
+- [improve][fn] Provide default error handler for function log appender [15728](https://github.com/apache/pulsar/pull/15728)
+- [fix][function] Fix error when user starts with the pulsar functions local runner [16565](https://github.com/apache/pulsar/pull/16565)
+- [fix][function] Fix python instance not process zip file correctly [16697](https://github.com/apache/pulsar/pull/16697)
+
+#### IO
+- [fix][io] throw exceptions when kafka offset backing store failed to start [14491](https://github.com/apache/pulsar/pull/14491)
```
Comment (momo-jun) [inline site2/website-next/release-notes/versioned/pulsar-2.8.4.md:116]:
```suggestion
- [fix][io] throw exceptions when Kafka offset backing store failed to start [14491](https://github.com/apache/pulsar/pull/14491)
```

File: site2/website-next/release-notes/versioned/pulsar-2.8.4.md [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+- [fix][io] Handle kafka sinks that return immutable maps as configs [14780](https://github.com/apache/pulsar/pull/14780)
```
Comment (momo-jun) [inline site2/website-next/release-notes/versioned/pulsar-2.8.4.md:117]:
```suggestion
- [fix][io] Handle Kafka sinks that return immutable maps as configs [14780](https://github.com/apache/pulsar/pull/14780)
```

File: site2/website-next/release-notes/versioned/pulsar-2.8.4.md [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+- [fix][sec] Fix multiple roles authorization [16645](https://github.com/apache/pulsar/pull/16645)
+- [fix][sec] Add timeout of sync methods and avoid call sync method for AuthoriationService (#15694) [16831](https://github.com/apache/pulsar/pull/16831)
+- [improve][sec] Support for get token from HTTP params [16871](https://github.com/apache/pulsar/pull/16871)
+- [improve][sec] Adapt basic authentication configuration with prefix [16935](https://github.com/apache/pulsar/pull/16935)
+
+#### Admin
+- [improve][admin] Add corresponding get command for namespace [12322](https://github.com/apache/pulsar/pull/12322)
+- [fix][admin] Fix deleting namespace will fail when system topics exist [15040](https://github.com/apache/pulsar/pull/15040)
+- [fix][admin] Fix validateGlobalNamespaceOwnership wrap exception issue [14612](https://github.com/apache/pulsar/pull/14612)
+- [fix][admin] Fix wrong prompt exception when get non-persistent topic list without GET_BUDNLE permission [14638](https://github.com/apache/pulsar/pull/14638)
```
Comment (momo-jun) [inline site2/website-next/release-notes/versioned/pulsar-2.8.4.md:97]:
```suggestion
- [fix][admin] Fix wrong prompt exception when getting the non-persistent topic list without GET_BUDNLE permission [14638](https://github.com/apache/pulsar/pull/14638)
```

File: site2/website-next/release-notes/versioned/pulsar-2.8.4.md [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+- [improve][broker] Avoid reconnection when a partitioned topic was created concurrently [16867](https://github.com/apache/pulsar/pull/16867)
+- [improve][broker] Recycle OpReadEntry in some corner cases [16869](https://github.com/apache/pulsar/pull/16869)
+- [fix][broker] Fix compaction subscription acknowledge Marker msg issue. [16918](https://github.com/apache/pulsar/pull/16918)
+
+#### Security
+- [fix][sec] Use provider's canLookupAsync method for AuthorizationService [11777](https://github.com/apache/pulsar/pull/11777)
+- [improve][sec] Optimize the logic of allowing namespace operation [13090](https://github.com/apache/pulsar/pull/13090)
+- [improve][sec] Full-support set ssl provider, ciphers and protocols [13740](https://github.com/apache/pulsar/pull/13740)
+- [fix][sec] Role with namespace produce authz can also get topics [13773](https://github.com/apache/pulsar/pull/13773)
+- [improve][sec] Full-support ssl provider, ciphers and protocols for broker service and proxy service [15034](https://github.com/apache/pulsar/pull/15034)
```
Comment (momo-jun) [inline site2/website-next/release-notes/versioned/pulsar-2.8.4.md:76]:
```suggestion
- [improve][sec] Full-support SSL provider, ciphers, and protocols for broker service and proxy service [15034](https://github.com/apache/pulsar/pull/15034)
```

File: site2/website-next/release-notes/versioned/pulsar-2.8.4.md [hunk 8]
```diff
@@ -0,0 +1,179 @@
...
+- [fix][test] Fix flaky AdminApiSchemaTest#testSchemaInfoApi [14508](https://github.com/apache/pulsar/pull/14508)
+- [fix][test] Fix NamespacesTest execution order [14552](https://github.com/apache/pulsar/pull/14552)
+- [fix][test] Fix AdminApiTest.testNamespaceSplitBundleConcurrent [14565](https://github.com/apache/pulsar/pull/14565)
+- [fix][test] Fix flaky test MemoryLimitTest#testRejectMessages (#14220) [14628](https://github.com/apache/pulsar/pull/14628)
+- [fix][ci] Upgrade zlib version to 1.2.12 [14964](https://github.com/apache/pulsar/pull/14964)
+- [improve][build] Ignore case when obfuscating passwords in configuration scripts [15077](https://github.com/apache/pulsar/pull/15077)
+- [fix][build] Fix LICENSE files for branch-2.8 [15261](https://github.com/apache/pulsar/pull/15261)
+- [Improve][doc] Add config of IO and acceptor threads in proxy [15340](https://github.com/apache/pulsar/pull/15340)
+- [improve][build] Configure DLog Bookie, Pulsar, and Admin clients via pass through config [15818](https://github.com/apache/pulsar/pull/15818)
```
Comment (momo-jun) [inline site2/website-next/release-notes/versioned/pulsar-2.8.4.md:160]:
```suggestion
- [improve][build] Configure DLog Bookie, Pulsar, and Admin clients via pass-through config [15818](https://github.com/apache/pulsar/pull/15818)
```

File: site2/website-next/release-notes/versioned/client-cpp-2.8.4.md [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+- [fix][cpp] Fix thread safety issue for multi topic consumer  [14380](https://github.com/apache/pulsar/pull/14380)
+- [fix][cpp] Fix wrong unit of Access Token Response's `expires_in` field [14554](https://github.com/apache/pulsar/pull/14554)
+- [fix][cpp] Handle exception in creating socket when fd limit is reached [14587](https://github.com/apache/pulsar/pull/14587)
+- [cleanup][cpp] Add braces around initialization of subobject [14735](https://github.com/apache/pulsar/pull/14735)
+- [fix][cpp] Fix producer is never destructed until client is closed [14797](https://github.com/apache/pulsar/pull/14797)
+- [fix][cpp] Fix flaky tests about reference count [14854](https://github.com/apache/pulsar/pull/14854)
+- [fix][cpp] Fix segmentation fault when creating socket failed [14834](https://github.com/apache/pulsar/pull/14834)
+- [fix][cpp] Fix the race condition of connect timeout task [14823](https://github.com/apache/pulsar/pull/14823)
+- [fix][cpp] Fix send callback might not be invoked in key based batching [14898](https://github.com/apache/pulsar/pull/14898)
```
Comment (momo-jun) [inline site2/website-next/release-notes/versioned/client-cpp-2.8.4.md:17]:
```suggestion
- [fix][cpp] Fix send callback might not be invoked in key-based batching [14898](https://github.com/apache/pulsar/pull/14898)
```

**Comment by BewareMyPower** [comment]
Fixed, PTAL again. @Anonymitaet @momo-jun

**Review by momo-jun — APPROVED** [review]

**Pull request merged**